bloom_test.go

```diff
@@ -0,0 +1,170 @@
+package bloom
+
+import (
+	"reflect"
+	"testing"
+)
+
+var bitTests = []struct {
+	f   *Filter
+	ins []int
+}{
+	{&Filter{}, []int{}},
+	{&Filter{f: []byte{0}}, make([]int, 0)},
+	{&Filter{f: []byte{1}}, []int{0}},
+	{&Filter{f: []byte{2}}, []int{1}},
+	{&Filter{f: []byte{3}}, []int{0, 1}},
+	{&Filter{f: []byte{0, 0}}, make([]int, 0)},
+	{&Filter{f: []byte{1, 0}}, []int{0}},
+	{&Filter{f: []byte{2, 0}}, []int{1}},
+	{&Filter{f: []byte{3, 0}}, []int{0, 1}},
+	{&Filter{f: []byte{0, 1}}, []int{8}},
+	{&Filter{f: []byte{0, 2}}, []int{9}},
+	{&Filter{f: []byte{0, 3}}, []int{8, 9}},
+	{&Filter{f: []byte{255}}, []int{0, 1, 2, 3, 4, 5, 6, 7}},
+	{&Filter{f: []byte{72, 97, 80, 130, 1, 8, 0, 4}}, []int{3, 6, 8, 13, 14, 20, 22, 25, 31, 32, 43, 58}},
+}
+
+func TestBit(t *testing.T) {
+	for _, test := range bitTests {
+		m := make(map[int]struct{})
+		for _, n := range test.ins {
+			m[n] = struct{}{}
+		}
+		for n := 0; n < len(test.f.f)*8; n++ {
+			var want int
+			if _, ok := m[n]; ok {
+				want = 1
+			}
```

Comment:
This might be a little more readable if you make it a map of ints. Roughly:

```
m := make(map[int]int)
for _, n := range test.ins {
	m[n] = 1
}
for n := 0; n < len(test.f.f)*8; n++ {
	want := m[n]
	...
```